eth/transactionManager.go

```diff
@@ -0,0 +1,259 @@
+
+	suggestedGasPrice := tm.gpm.GasPrice()
+
+	// If the suggested gas price is higher than the bumped gas price, use the suggested gas price
+	// This is to account for any wild market gas price increases between the time of the original tx submission and time
+	// of replacement tx submission
+	// Note: If the suggested gas price is lower than the bumped gas price because market gas prices have dropped
+	// since the time of the original tx submission we cannot use the lower suggested gas price and we still need to use
+	// the bumped gas price in order to properly replace a still pending tx
+	if suggestedGasPrice.Cmp(gasPrice) == 1 {
+		gasPrice = suggestedGasPrice
+	}
+
+	// Bump gas price exceeds max gas price, return early
+	max := tm.gpm.MaxGasPrice()
+	if gasPrice.Cmp(max) > 0 {
+		return nil, fmt.Errorf("Replacement gas price exceeds max gas price suggested=%v max=%v", gasPrice, max)
+	}
+
+	// Replacement raw tx uses same fields as old tx (reusing the same nonce is crucial) except the gas price is updated
+	newRawTx := types.NewTransaction(tx.Nonce(), *tx.To(), tx.Value(), tx.Gas(), gasPrice, tx.Data())
+
+	newSignedTx, err := tm.sig.SignTx(newRawTx)
+	if err != nil {
+		return nil, err
+	}
+
+	sendErr := tm.eth.SendTransaction(context.Background(), newSignedTx)
+	txLog, err := newTxLog(tx)
+	if err != nil {
+		txLog.method = "unknown"
+	}
+	if sendErr == nil {
+		glog.Infof("\n%vEth Transaction%v\n\nReplacement transaction: \"%v\".  Gas Price: %v \nTransaction Failed: %v\n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, newSignedTx.GasPrice().String(), err, strings.Repeat("*", 75))
+	} else {
+		glog.Infof("\n%vEth Transaction%v\n\nReplacement transaction: \"%v\".  Hash: \"%v\".  Gas Price: %v \n\n%v\n", strings.Repeat("*", 30), strings.Repeat("*", 30), txLog.method, newSignedTx.Hash().String(), newSignedTx.GasPrice().String(), strings.Repeat("*", 75))
+	}
+
+	return newSignedTx, sendErr
+}
+
+func (tm *TransactionManager) checkTxLoop() {
+	for {
+		tm.cond.L.Lock()
+		for tm.queue.length() == 0 {
+			tm.cond.Wait()
+
+			select {
+			case <-tm.quit:
+				tm.cond.L.Unlock()
+				glog.V(common.DEBUG).Info("Stopping transaction manager")
+				return
+			default:
+			}
+		}
+
+		tx := tm.queue.pop()
+		tm.cond.L.Unlock()
+
+		var (
+			receipt *types.Receipt
+			err     error
+		)
+
+		receipt, err = tm.wait(tx)
+
+		// context.DeadlineExceeded indicates that we hit the txTimeout
+		// If we hit the txTimeout, replace the tx up to maxReplacements times
+		for i := 0; err == context.DeadlineExceeded && i < tm.maxReplacements; i++ {
+			tx, err = tm.replace(tx)
+			// Do not attempt additional replacements if there was an error submitting this
+			// replacement tx
+			if err != nil {
+				break
+			}
+			receipt, err = tm.wait(tx)
+		}
+
+		tm.feed.Send(&transactionReceipt{
+			Receipt: receipt,
+			err:     err,
+		})
+
+	}
+}
+
+func calcReplacementGasPrice(tx *types.Transaction) *big.Int {
```

Comment:
We don't need to address this in this PR, but just making a note that we will probably want to eventually make the gas price bump a parameter given that its a value that influences how soon a replacement tx will be mined and given that the gas price bump required by an ETH node technically doesn't have to be 10% (that just happens to be the default on geth, but there is an option to customize it).

Reply:
Wouldn't that only be supportable by self-hosted geth nodes because if the configuration on the livepeer node is set to a lower replacement bump than i.e. infura you would still see `replacement tx underpriced` no ?

Reply:
Yeah if your gas price bump is lower than the gas price bump required by the ETH provider you would receive a `replacement tx underpriced` error.

Aside from speeding up the confirmation of replacement txs, the motivation for supporting custom gas price bumps is to give the livepeer node operator to adjust the gas price bump (i.e. increase it) used if the ETH provider requires a gas price bump different (i.e. greater) than the current default of 10%. If the node operator does not control the ETH provider (i.e. self-hosted node(s)) and the ETH provider is configured to use a non-default gas price bump (or if the default ever changes) then the current hardcoded gas price bump in the livepeer node may not work.

Anyways this isn't that crucial at the moment, but just wanted to point this out for future reference.